src/main/java/uk/gov/hmcts/reform/civil/service/search/BundleCreationTriggerService.java

```diff
@@ -0,0 +1,39 @@
+package uk.gov.hmcts.reform.civil.service.search;
+
+import org.elasticsearch.index.query.QueryBuilder;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.civil.enums.CaseState;
+import uk.gov.hmcts.reform.civil.model.search.Query;
+import uk.gov.hmcts.reform.civil.service.CoreCaseDataService;
+
+import java.time.LocalDate;
+import java.util.List;
+
+import static org.elasticsearch.index.query.QueryBuilders.boolQuery;
+import static org.elasticsearch.index.query.QueryBuilders.matchQuery;
+import static org.elasticsearch.index.query.QueryBuilders.rangeQuery;
+
+@Service
+public class BundleCreationTriggerService extends ElasticSearchService{
+
+    private static final int BUNDLE_CREATION_TIME_RANGE = 3;
+    public BundleCreationTriggerService(CoreCaseDataService coreCaseDataService){
+        super(coreCaseDataService);
+    }
+
+    public Query query(int startIndex){
+        return new Query(
+            boolQuery()
+                .minimumShouldMatch(1)
+                .should(boolQuery()
+                            .must(rangeQuery("data.hearingDate").lt(LocalDate.now().plusWeeks(BUNDLE_CREATION_TIME_RANGE)))
```

Comment:
Does your query need to check this eventuality as well?:
`IF hearing date moves after the 3 weeks before a hearing date
And a bundle has already been created
And the "Order" has been flagged as needing a new bundle
Then a new bundle is created and displayed on a tab <Location TBC> in date order with any previous court bundle`
Or can you take care of this somewhere else?

Reply:
@suvarnathekkoot has explained that this element is dependent on ticket https://tools.hmcts.net/jira/browse/CIV-6041 to finish creating a tab.

Will resolve conversation.